cmd/keeper/cmd/keeper.go

```diff
@@ -1812,25 +1826,35 @@ func keeper(c *cobra.Command, args []string) {
 		die("cannot create data dir: %v", err)
 	}
 
-	if cfg.pgListenAddress == "" {
-		die("--pg-listen-address is required")
+	if cfg.pgAdvertiseAddress == "" && cfg.pgListenAddress == "" {
+		die("--pg-listen-address and/or --pg-advertise-address is required")
+	} else if cfg.pgListenAddress == "" {
+		cfg.pgListenAddress = "*"
+	} else if cfg.pgAdvertiseAddress == "" {
+		cfg.pgAdvertiseAddress = cfg.pgListenAddress
 	}
 
-	ip := net.ParseIP(cfg.pgListenAddress)
+	if cfg.pgAdvertisePort == "" && cfg.pgPort == "" {
+		die("--pg-port and/or --pg-advertise-port is required")
+	} else if cfg.pgPort == "" {
+		cfg.pgPort = cfg.pgAdvertisePort
+	} else if cfg.pgAdvertisePort == "" {
+		cfg.pgAdvertisePort = cfg.pgPort
+	}
```

Comment:
This looks strange since if pgPort is not specified it'll be set to advertise port and if pgAdvertisePort is not specified it'll be set to pgPort. But pgPort has a default to 5432 so it'll probably never happen.

I'll prefer to keep the original behavior and just set pgAdvertisePort to pgPort when pgAdvertisePort is not specified.

Reply:
ok, no problem :)